governor/pilot/probe_sample.py

```
@@ -0,0 +1,22 @@
+Standalone review-bait for an external-reviewer probe round. Nothing
+imports this file. Both functions deliberately contradict their own
+docstrings; the defects are removed in the follow-up commit.
+"""
+
+
+def is_head_fresh(recorded_head: str, current_head: str) -> bool:
+    """Return True iff the recorded head still equals the current head."""
+    return recorded_head != current_head
+
+
+def unique_short_shas(shas: list, seen=[]) -> list:
```

Ruff (0.16.1)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)